src/differentials/composite.jl

```diff
@@ -36,6 +36,12 @@ function Composite{P}(args...) where P
     return Composite{P, typeof(args)}(args)
 end
 
+function Composite{P}(d::Dict) where {P<:Dict}
```

Comment:
This doesn't actually match the signature for `Composite` of TUples or NamedTuples.
If you want to pass the actual object in the signature they use is `Composite{P, T}(obj::T)`

I am tempted to say we should change that at some point, but that is a different PR

Reply:
Hmmm good point. I'm not entirely sure what the right way to make this consistent with the existing signatures would be though -- some kind of collection of `Pair`s perhaps?

Or possibly I could mirror the constructors in `Base`?